internal/guest/runtime/runc/container.go

```diff
@@ -363,9 +366,17 @@ func (c *container) startProcess(
 	}
 	args = append(args, c.id)
 
+	if annotations != nil {
+		_, exists := annotations["io.microsoft.bmc.logging.pipelocation"]
+		if exists {
+			args = append(args, "--preserve-fds", "2")
```

Comment:
This can be put down near line 462, and just append to `cmd.Args`. That will ensure we only pass `--preserve-fds` for the sidecar container.